server/src/test/workspaceModules.test.ts

```diff
@@ -0,0 +1,149 @@
+import { expect } from "chai";
+import { stub } from "sinon";
+import { CheckFileResult, OMTModule } from "../types";
+import { WorkspaceModules } from "../workspaceModules";
+
+
+describe('WorkspaceModule', () => {
+    let workspaceModules: WorkspaceModules;
+    let defaultCheckFileResult: CheckFileResult;
+    beforeEach(() => {
+        workspaceModules = new WorkspaceModules();
+        defaultCheckFileResult = {
+            path: 'folder/defaultPath.omt',
+            module: {
+                name: 'defaultModule',
+            }
+        };
+    });
+
+    describe('checkForChanges', () => {
+        it('should ignore it if the result is not a module', () => {
+            workspaceModules.checkForChanges({
+                ...defaultCheckFileResult,
+                module: undefined
+            });
+            expect(workspaceModules.modules.size).to.eq(0);
+        });
+
+        it('should create a module when the name is new', () => {
+            workspaceModules.checkForChanges(defaultCheckFileResult);
+            expect(workspaceModules.modules.size).to.eq(1);
+            const module = <OMTModule>workspaceModules.modules.values().next().value;
+            expect(module.name).to.eq(defaultCheckFileResult.module?.name);
+            expect(module.uri).to.eq(defaultCheckFileResult.path);
+        });
+
+        it('should replace an existing module with the same name but a different path', () => {
```

Comment:
The previous path was never loaded during this test, so this tests the exact same thing as the previous test. You'll have to check for changes twice for an update to happen.

Reply:
added another `checkForChanges`